apps/prairielearn/src/pages/instructorFileEditor/instructorFileEditor.js

```diff
@@ -143,11 +144,22 @@ router.get('/*', (req, res, next) => {
       }
 
       if ('jobSequence' in fileEdit) {
-        fileEdit.jobSequence.jobs.forEach((item) => {
-          if (item.type === 'git_push' && item.status === 'Error') {
+        // New case: single job for entire operation. We inspect the job's
+        // output to determine if pushing failed.
+        if (!fileEdit.jobSequence.legacy) {
+          const job = fileEdit.jobSequence.jobs[0];
+          if (job.output_raw.indexOf('error: failed to push some refs') >= 0) {
```

Comment:
I definitely don't like this as much as the previous multiple-job implementation, as this only handles one kind of failure. A few thoughts:

- We can allow modern server jobs to record some metadata about their own execution?
- We can automatically track each command that is executed, its arguments, exit code, execution time, etc. and then query that directly.

Reply:
I think the implementation would be an extra `metadata` (bikeshed away) `jsonb` column, maybe with a `commands` array? This would track each command. This means we'd have to hunt through it with something like `commands.find(c => c.file === 'git' && c.args[0] === 'push')`. I think this is reasonable?

Reply:
As it turns out, I accidentally broke the editor error page with my other recent change, which also relied on being able to inspect individual jobs in the old implementation:

https://github.com/PrairieLearn/PrairieLearn/blob/2d2a2c9dcf4d54fae6af91382a8e7d69f62093f3/apps/prairielearn/src/pages/editError/editError.js#L31-L32

So we'll definitely need to do something about this :(

Reply:
I attempted to improve this in 691ab8e. It looks like tests are failing though; I'll revisit this soon.

Reply:
Fixed things up in 639a2aa! See https://github.com/PrairieLearn/PrairieLearn/pull/7966#discussion_r1244479860 for a small discussion on this.